internal/services/web/static_site_resource.go

```diff
@@ -115,14 +115,21 @@ func resourceStaticSiteCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{
 
 	loc := location.Normalize(d.Get("location").(string))
 
+	skuName := d.Get("sku_size").(string)
+
 	identity, err := expandStaticSiteIdentity(d.Get("identity").([]interface{}))
 	if err != nil {
 		return fmt.Errorf("expanding `identity`: %+v", err)
 	}
 
+	// See: https://github.com/Azure/azure-rest-api-specs/issues/17525
+	if skuName == string(web.SkuNameFree) && identity != nil {
+		return fmt.Errorf("Free tier Static Site doens't support specifying `identity`")
```

Comment:
this is probably more descriptive as?

```suggestion
		return fmt.Errorf("a Managed Identity cannot be used when tier is set to `Free`")`
```